lib/ansible/modules/cloud/amazon/elb_application_lb.py

```diff
@@ -100,12 +100,20 @@
     description:
       - A dictionary of one or more tags to assign to the load balancer.
     required: false
+  type:
+    description:
+      - Type of Load Balaner, Application or Network Load Balancer.
+    required: false
+    default: application
+    choices: ['application', 'network']
```

Comment:
Hm. Since this module is for application load balancers, using it to make a network load balancer might be confusing/overload the module. Would a new elb_network_lb module make sense? A possibility would be to make a load_balancer directory in module_utils/aws/ and move common code into that.

Reply:
This probably makes sense.  Code is already duplicated across the _facts modules anyway.

Reply:
I've started work here to create a separate nlb module and move common code https://github.com/wimnat/ansible/tree/feature/elb_network_lb

Reply:
@wimnat Are you able to work on elb_network_lb for 2.5? If you have enough on your plate already I'll work on top of your start and open a PR.